chore(tabs): match mocks

jinlee93 opened the pull request:
### Summary:
- matches tabs [mocks](https://www.figma.com/file/DEm9VhynEYAlx66ogEuskp/Tab-Navigation---Component-Exploration-%2B-Creation?node-id=3870%3A10986)
  - adjusts tabs font
  - adjusts active indicator box size
  - adds scroll indicators


https://user-images.githubusercontent.com/86632227/175191559-76d83b5c-e32c-412f-9d15-c9b3f2d15883.mov


### Test Plan:
- unit tests pass
- no visual regression except those using tabs

[sc-202249]

jinlee93 commented on src/components/Tabs/Tabs.module.css:
```diff
@@ -5,12 +5,57 @@
+  top: 0;
+  width: var(--eds-size-8);
+
+  height: 2.75rem;
```
Height needs to be fixed since `content: ''`. Alternative is to have a `calc(padding + text-size * line-height multiplier)` but 1) text sizes and line-height multipliers are not css variables/tokens anyways and 2) figured that would be more confusing than having fixed height.

jinlee93 replied:
Is it possible to have `--current-font-size` and `--current-line-height` declared within typography tokens for use in cases like this, or would that be too niche?

jinlee93 commented on src/components/Tabs/Tabs.stories.tsx:
```diff
@@ -40,7 +40,79 @@ export default {
+          </Text>
+        </Tab>
+
+        <Tab title="Tab Title 9">
```
added more tab components in the tabs to display scroll functionality. Maybe looks cleaner with some array looping but this is more explicit especially with how the `Tab` component works

jinlee93 commented on src/components/Tabs/Tabs.tsx:
```diff
@@ -181,6 +185,48 @@ export const Tabs = ({
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        handleTabsScroll(headerRef.current!);
+      };
+      resizeHandleTabs();
```
The event listener actually calls the callback once when initiated, but the event listener is not triggered with prop changes so this line is required.
Sure it may be called twice on initial paint but that's better than it not being called at all.

dierat replied:
Maybe add something to that effect as a comment? Tracking down bugs associated with react lifecycle event triggers can be pretty annoying.

jinlee93 replied:
Makes sense to me, will add

dierat commented:
How do I select another tab using a keyboard? I only see a focus indicator on the active tab; the arrow keys scrolls the container of tabs and I don't know where the focus goes after clicking the "tab" key 🤔

dierat commented:
>How do I select another tab using a keyboard? I only see a focus indicator on the active tab; the arrow keys scrolls the container of tabs and I don't know where the focus goes after clicking the "tab" key 🤔

Looks like this is an existing issue; I'm seeing the same behavior on `next`. I guess there's an error in `onKeyDown`, probably a similar one to the `Popover` keyboard bug. Once that's fixed, presumably you would scroll the tabs via keyboard by using the arrow keys to select tabs that are off-screen or something.

jinlee93 commented:
> > How do I select another tab using a keyboard? I only see a focus indicator on the active tab; the arrow keys scrolls the container of tabs and I don't know where the focus goes after clicking the "tab" key 🤔
> 
> Looks like this is an existing issue; I'm seeing the same behavior on `next`. I guess there's an error in `onKeyDown`, probably a similar one to the `Popover` keyboard bug. Once that's fixed, presumably you would scroll the tabs via keyboard by using the arrow keys to select tabs that are off-screen or something.

Yea think that was an existing issue, as I didn't touch functionality outside of the fade indicators. I'll fix the arrow key to select tabs bug in this PR as well

dierat commented on src/components/Tabs/Tabs.module.css:
```diff
@@ -5,12 +5,57 @@
+ */
+.tabs--scrollable-left::before {
+  left: 0;
+  background-image: linear-gradient(
+    to right,
+    var(--eds-theme-color-background-neutral-default),
+    transparent 90%
+  );
```
Another thing you could try is a mask with a linear gradient on the `tabs__list` (or the header). That way, it would fade into the background no matter what the background color is behind the tabs, not just on white backgrounds. https://stackoverflow.com/questions/60996982/css-opacity-gradient-top-to-bottom-without-color (Feel free to ignore this comment or make a separate task if it feels like too much work right now; it's not blocking.)

jinlee93 replied:
I think using mask is a more valid approach. Will try it, thanks!

jinlee93 replied:
Definitely looks a lot better, thanks!


https://user-images.githubusercontent.com/86632227/175438883-5eb81d9b-1c4d-4b9e-81c2-dfadc9f0f87f.mov


Now there is a visual bug with the mask covering the scroll bar as well, but I'm going to file a separate [backlog ticket](https://app.shortcut.com/czi-edu/story/202962) for it 

![Screen Shot 2022-06-23 at 4 51 02 PM](https://user-images.githubusercontent.com/86632227/175438996-05aa7366-cec6-49db-ab40-08181f356ec6.png)
![Screen Shot 2022-06-23 at 5 47 57 PM](https://user-images.githubusercontent.com/86632227/175439000-4fd6a057-8549-498e-804b-14b73b1e9636.png)

jinlee93 replied:
Yup will do!

jinlee93 commented:
> Looks like this is an existing issue; I'm seeing the same behavior on next. I guess there's an error in onKeyDown, probably a similar one to the Popover keyboard bug. Once that's fixed, presumably you would scroll the tabs via keyboard by using the arrow keys to select tabs that are off-screen or something.

> Yea think that was an existing issue, as I didn't touch functionality outside of the fade indicators. I'll fix the arrow key to select tabs bug in this PR as well

Fixed and added test for it

dierat commented on src/components/Tabs/Tabs.test.tsx:
```diff
@@ -1,6 +1,35 @@
+
 describe('<Tabs />', () => {
   generateSnapshots(stories);
+  it('should focus and select with keyboard controls', () => {
```
This is great, thanks!

dierat reviewed (approved):
Thanks for fixing the functionality while you were in here!
